### ISSUE TITLE
Does not build with MSVS in Debug configuration

### ISSUE THREAD

**Issue by PBfordev** [Blake-Madden/Wisteria-Dataviz#2]
When attempting to build x64 Debug configuration using generated MSVS solution, I get the error:
```
Building Custom Rule D:/Dev/Desktop/!Lib/Wisteria-Dataviz-GIT/CMakeLists.txt
2>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(700,5): error MSB6006: "CL.exe" exited with code 2.
2>cl : command line error D8016: '/O2' and '/RTC1' command-line options are incompatible
2>Done building project "Wisteria.vcxproj" -- FAILED.
```
The Release configuration builds just fine. I built only a couple of libraries with CMake but never encountered this error.

Can you really use `CMAKE_BUILD_TYPE` like this with multi-config generators:
https://github.com/Blake-Madden/Wisteria-Dataviz/blob/da306a17697cc226b9b025f231a3ce936985f297/CMakeLists.txt#L81-L99

MSVS 2022 Community 17.1.4
CMake 3.22
wxWidgets current master

**Comment by Blake-Madden** [comment]
In VS, if you go to Project -> CMake settings, is the "Configuration Type" set to "Debug"?  You can send me your CMakeSettings.json file and I can take a look at it.

**Comment by PBfordev** [comment]
> In VS, if you go to Project -> CMake settings, is the "Configuration Type" set to "Debug"? You can send me your CMakeSettings.json file and I can take a look at it.

Sorry, I never used CMake in MSVS and cannot find  CMake settings anywhere in MSVS, be it in the menu or project properties dialog. There is "CMake Workspace Settings" at the bottom of the Project menu but when I click it nothing happens, regardless of which project is selected in the Solution Explorer.

I also could not find CMakeSettings.json: I certainly did not create any such file nor is any .json anywhere in the build dir. 

I just used CMakeGUI to generate "Visual Studio 17 2022" project, as I have always done before with other projects without any issues. I believed this issue stemmed from the quoted `ELSE()` branch always being executed in multi-config generators, assuming in those the generator expressions must be used. But I admit I know little about CMake.

But again, this issue is easy to work around by changing the optimization option in the project properties manually.

**Comment by Blake-Madden** [comment]
Merged PR #4, thanks!
